consensus/core/src/synchronizer.rs

```diff
@@ -611,11 +611,25 @@ impl<C: NetworkClient, V: BlockVerifier, D: CoreThreadDispatcher> Synchronizer<C
             if let Err(e) = block_verifier.verify(&signed_block) {
                 // TODO: we might want to use a different metric to track the invalid "served" blocks
                 // from the invalid "proposed" ones.
+
+                // It's important to check the validity of the index, otherwise the method will panic.
+                let hostname = if context.committee.is_valid_index(signed_block.author()) {
```

Comment:
Can we pass the `peer_index` to `verify_blocks()` instead? The peer is the culprit here since it should not serve invalid blocks to others.

Reply:
That's good point. I'll switch to using the `peer_index`. We need to keep in mind then that for an invalid block report on the metrics we can see either:
1. the original author coming up
2. or the peer to served an invalid block

there is no good way to distinguish anyways unless we always report both, but for now might be good enough

Reply:
For timestamp checks, only the block authority is available, which should be equivalent to the peer here since the block has been verified. Or are there other places where peer is unavailable? We can take the opportunity to see if it is possible to fix them. For a few errors, the indicated author can be arbitrary so counting them in metrics is not very useful.

Reply:
> Or are there other places where peer is unavailable?

There shouldn't. At least from all the synchronization paths peer is available.